testscript/testscript_test.go

```diff
@@ -202,6 +202,58 @@ func TestTestwork(t *testing.T) {
 	}
 }
 
+// TestGoTmpDir tests that a non zero value in $GOTMPDIR is honoured
+func TestGoTmpDir(t *testing.T) {
+	// NOTE: do not make this test parallel; we change environment variables within it
+	td, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("failed to create temp dir: %v", err)
+	}
+	prev := os.Getenv("GOTMPDIR")
+	if err := os.Setenv("GOTMPDIR", td); err != nil {
+		t.Fatalf("failed to set GOTMPDIR: %v", err)
+	}
+	defer func() {
+		if err := os.Setenv("GOTMPDIR", prev); err != nil {
```

Comment:
I don't believe that `Setenv` can fail in this case. I wouldn't bother checking the error return, personally. (and ditto for the Setenv above).